pkg/credentialprovider/gcp/metadata.go

```diff
@@ -142,7 +145,17 @@ func onGCEVM() bool {
 
 // Enabled implements DockerConfigProvider for all of the Google implementations.
 func (g *MetadataProvider) Enabled() bool {
-	return onGCEVM()
+	onGCE := onGCEVM()
```

Comment:
Cross linking this thread in case it gets lost in the comments https://github.com/kubernetes/kubernetes/pull/102507#discussion_r673290425

I'm not following the logic here to switch the order back to run onGCEVM first. If the feature gate is enabled (i.e. disabled), then there's no reason to ever call onGCEVM, because the plugin is explicitly disabled. In this case, the warning message makes sense too, because the plugin is disabled anyways. When it is disabled, why bother running onGCEVM which is a relatively expensive operation that reads a file from disk?

Reply:
I think this got resolved below, right? this is to prevent logging GCE-specific messages when we're not even running on GCE, right?

Reply:
I believe so yes. The cloud provider plugins are explicitly enabled. However the credential provider plugins are all "enabled". So we want to early out of this plugin if we are not on GCE without a warning that this plugin is disabled. The disabled warning only makes sense if you are on GCE.